packages/worker/src/worker-tuner.ts

```diff
@@ -33,23 +44,41 @@ type ResourceBasedSlotsForType = ResourceBasedSlotOptions & {
  *
  * @experimental
  */
-export type SlotSupplier = ResourceBasedSlotsForType | FixedSizeSlotSupplier;
+export type SlotSupplier<SI extends SlotInfo> =
+  | ResourceBasedSlotsForType
+  | FixedSizeSlotSupplier
+  | CustomSlotSupplier<SI>;
+
+/**
+ * Resource based slot supplier options for a specific kind of slot.
+ *
+ * @experimental
+ */
+type ResourceBasedSlotsForType = ResourceBasedSlotOptions & {
+  type: 'resource-based';
+  tunerOptions: ResourceBasedTunerOptions;
+};
 
 /**
  * Options for a specific slot type within a {@link ResourceBasedSlotsForType}
  *
  * @experimental
  */
 export interface ResourceBasedSlotOptions {
-  // Amount of slots that will be issued regardless of any other checks.
-  // Defaults to 2 for workflow tasks and 1 for activity tasks.
+  /**
+   * Amount of slots that will be issued regardless of any other checks.
+   * Defaults to 2 for workflow tasks and 1 for activity tasks.
+   */
   minimumSlots?: number;
-  // Maximum amount of slots permitted
-  // Defaults to 1000 for workflow tasks and 2000 for activity tasks.
+  /**
+   * Maximum amount of slots permitted
+   * Defaults to 1000 for workflow tasks and 2000 for activity tasks.
```

Comment:
I know this is not new from this PR, but reading this again makes me think… Isn't 1000/2000 ridiculously high as defaults? Or will this effectively get capped by some other limit anyway (I mean, something else than only resource usage).

IIRC, the default namespace config in Cloud is to limit the _total_ number of polers to 2000. Could we get in a situation where the RB provider would allow that many slots in _polling state_ at the same time, from the same worker?

Should we have a distinct limit for the number of slots that have been allowed but haven't yet been "marked as in-use"?

Reply:
Yeah, it is, but the idea is it's supposed to be capped by the resource tuner long before it gets to that point. So, not something else, but that.

Good question about the polling, but right now no way, and when I do poller tuning I'll have to consider that of course. Probably there is also just a different limit on max polling ceiling which will in effect restrict this (or this will be bound to that)